### PR TITLE
accept more postal codes

### PR DESCRIPTION
**Pull request by aiquita**

### CODE REVIEW

**Comment by aiquita** [comment]
Can I still use staging db for test purposes?

**Comment by Schedulaar** [comment]
@maxammann https://tiles.staging.ehrenamtskarte.app/map.html gives a 502 Bad Gateway... Is the staging server still running?

**Comment by maxammann** [comment]
It wasnt running anymore, I started it again.

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/importer/steps/DataTransformation.kt [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+        val pc = if (store.postalCode != null) findPostalCode(store.postalCode) else ""
+        store.copy(
+            categoryId = if (store.categoryId == 99) 9 else store.categoryId,
+            postalCode = pc
         )
     }
 
+    companion object {
+
+        private const val POSTAL_CODE_LENGTH = 5
```
Comment (maxammann) [inline backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/importer/steps/DataTransformation.kt:21]: We could still support 4-long postal codes :thinking: 

Rolf is greeting! (see the 5 fingers :P)

image

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/importer/RawDataFilterBuilder.kt [hunk 2]
```diff
@@ -16,9 +17,7 @@ data class FilterBuilder(val store: LbeAcceptingStore, val logger: Logger) {
     fun filterPostalCode(): Boolean {
         val pc = store.postalCode
         val valid = store.postalCode == null
-                || store.postalCode?.trim()?.length == 5
-                || store.postalCode?.trim()?.length == 0 // must be 5 symbols or completely empty!
-                || pc != null && matchesNa(pc)
+                || findPostalCode(pc!!).length == 5
```
Comment (maxammann) [inline backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/importer/RawDataFilterBuilder.kt:20]: Should we reuse the constant defined below?

**Review by maxammann — APPROVED** [review]

**Comment by maxammann** [comment]
Should we close this or do we still want to do changes here?

**Comment by maxammann** [comment]
reopen if still relevant

**Pull request closed without merging**